File: src/csharp/api/Models/PatientOrderHistory.cs
using Example.Api.Enums;

namespace Example.Api.Models;

/// <summary>
/// Represents the history of a patient order.
/// </summary>
public record PatientOrderHistory
{
    /// <summary>
    /// The unique identifier for the patient order history record.
    /// </summary>
    /// <value></value>
    public long Id { get; set; }

    /// <summary>
    /// The type of log entry.
    /// </summary>
    /// <value></value>
    public LogType Type { get; set; }

    /// <summary>
    /// The patient order identifier associated with this history record.
    /// </summary>
    /// <value></value>
    public long OrderId { get; set; }

    /// <summary>
    /// The identifier of the patient associated with this order history.
    /// </summary>
    /// <value></value>
    public long PatientId { get; set; }

    /// <summary>
    /// The status of the patient order at this point in history.
    /// </summary>
    /// <value></value>
    public OrderStatus Status { get; set; }

    /// <summary>
    /// The remarks or notes associated with this history record.
    /// </summary>
    /// <value></value>
    public string? Remarks { get; set; }

    /// <summary>
    /// The user who logged this history record.
    /// </summary>
    /// <value></value>
    public string LogBy { get; set; } = default!;

    /// <summary>
    /// The timestamp when this history record was logged.
    /// </summary>
    /// <value></value>
    public DateTimeOffset LogAt { get; set; }
}

File: src/python/app/entities/patient_order_history.py
"""PatientOrderHistory entity model."""

from __future__ import annotations
from datetime import datetime
from app.entities.base import Base
from app.enums.log_type import LogType
from app.enums.order import OrderStatus
from app.db.types import IntEnumType
from sqlalchemy import DateTime, ForeignKey, Index, String
from sqlalchemy import Integer
from sqlalchemy.orm import Mapped, mapped_column


class PatientOrderHistory(Base):
    """Patient order history model."""

    __tablename__ = "patient_order_history"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    type: Mapped[LogType] = mapped_column(IntEnumType(LogType), nullable=False, default=LogType.Add)
    order_id: Mapped[int] = mapped_column(ForeignKey("patient_order.id", ondelete="RESTRICT"), nullable=False)
    patient_id: Mapped[int] = mapped_column(Integer, nullable=False)
    status: Mapped[OrderStatus] = mapped_column(IntEnumType(OrderStatus), nullable=False)
    remarks: Mapped[str] = mapped_column(String(255), nullable=True)
    log_by: Mapped[str] = mapped_column(String(50), nullable=False)
    log_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
